packages/apollo-gateway/src/__tests__/integration/nockMocks.ts

```diff
@@ -0,0 +1,114 @@
+import nock from 'nock';
+
+export const mockLocalhostSDLQuery = () =>
+  nock('http://localhost:4001', { encodedQueryParams: true })
+    .post('/graphql', {
+      query: 'query GetServiceDefinition { _service { sdl } }',
+    })
+    .reply(
+      200,
+      {
+        data: {
+          _service: {
+            sdl:
+              'extend type Query {\n  me: User\n  everyone: [User]\n}\n\ntype User @key(fields: "id") {\n  id: ID!\n  name: String\n  username: String\n}\n',
+          },
+        },
+      },
+      {
+        'Content-Type': 'application/json',
+      },
+    );
+
+export const mockFetchStorageSecret = () =>
+  nock('https://storage.googleapis.com:443', { encodedQueryParams: true })
+    .get(
+      '/engine-partial-schema-prod/mdg-private-6077/storage-secret/232d3a78a4392a756d37613458805dd830b517453729dc4945c24ca4b575cfc270b08e763cea5a8da49c784b31272d1c79b1d598cdc6e4d6913700a5505cd2f7.json',
+    )
+    .reply(
```

Comment:
What is going on here? This looks like a buffer perhaps? I think we should try and make it clearer what is happening in this test helper

Reply:
Remade the mock installation functions.